packages/block-editor/src/utils/dom.js

```diff
@@ -140,15 +140,37 @@ export function getVisibleElementBounds( element ) {
 		return new window.DOMRectReadOnly();
 	}
 
-	let bounds = element.getBoundingClientRect();
+	const parentRect = element.getBoundingClientRect();
+	let bounds = parentRect;
 
 	const stack = [ element ];
+	const xValues = [];
+	const yValues = [];
 	let currentElement;
 
 	while ( ( currentElement = stack.pop() ) ) {
 		for ( const child of currentElement.children ) {
 			if ( isElementVisible( child ) ) {
 				const childBounds = child.getBoundingClientRect();
+				yValues.push( childBounds.y );
+				xValues.push( childBounds.x );
+				/*
+				 * If the child is larger than the parent, adjust the x and y values to the greatest known
+				 * to account for any scrolling.
+				 */
+				if ( childBounds.width > parentRect.width ) {
+					childBounds.x = Math.max(
+						parentRect.x,
+						xValues.sort().at( -1 ) || 0
```

Comment:
Just a thought, but if we stick with the idea of caching values, do we need arrays? Could we set `let maxX = 0` and only update `maxX` if `childBounds.x` is greater than it?